kratos/solving_strategies/schemes/residual_based_bdf_displacement_scheme.h

```diff
@@ -283,91 +266,38 @@ class ResidualBasedBDFDisplacementScheme
-                    dotun0[2] /= BDFBaseType::mBDF[0];
-
-                    un0[2] = dotun0[2];
-                    for (std::size_t i_order = 1; i_order < BDFBaseType::mOrder + 1; ++i_order)
-                        un0[2] -= BDFBaseType::mBDF[i_order] * it_node->FastGetSolutionStepValue(DISPLACEMENT_Z, i_order);
-                    un0[2] /= BDFBaseType::mBDF[0];
-                    predicted_z = true;
+            if (accelpos > -1) {
+                for (std::size_t i_dim = 0; i_dim < dimension; ++i_dim) {
+                    if (it_node->GetDof(accel_components[i_dim], accelpos + i_dim).IsFixed()) {
+                        dotun0[i_dim] = dot2un0[i_dim];
+                        for (std::size_t i_order = 1; i_order < BDFBaseType::mOrder + 1; ++i_order)
+                            dotun0[i_dim] -= BDFBaseType::mBDF[i_order] * it_node->FastGetSolutionStepValue(vel_components[i_dim], i_order);
+                        dotun0[i_dim] /= BDFBaseType::mBDF[i_dim];
+
+                        un0[i_dim] = dotun0[i_dim];
+                        for (std::size_t i_order = 1; i_order < BDFBaseType::mOrder + 1; ++i_order)
+                            un0[i_dim] -= BDFBaseType::mBDF[i_order] * it_node->FastGetSolutionStepValue(disp_components[i_dim], i_order);
+                        un0[i_dim] /= BDFBaseType::mBDF[i_dim];
+                        predicted[i_dim] = true;
+                    }
                 }
             }
-            if (velpos_z > -1 && !predicted_z) {
-                if (it_node->GetDof(VELOCITY_Z, velpos_z).IsFixed() && !predicted_z) {
-                    un0[2] = dotun0[2];
-                    for (std::size_t i_order = 1; i_order < BDFBaseType::mOrder + 1; ++i_order)
-                        un0[2] -= BDFBaseType::mBDF[i_order] * it_node->FastGetSolutionStepValue(DISPLACEMENT_Z, i_order);
-                    un0[2] /= BDFBaseType::mBDF[0];
-                    predicted_z = true;
+            if (velpos > -1) {
+                for (std::size_t i_dim = 0; i_dim < dimension; ++i_dim) {
+                    if (it_node->GetDof(vel_components[i_dim], velpos + i_dim).IsFixed() && !predicted[i_dim]) {
+                        un0[i_dim] = dotun0[i_dim];
+                        for (std::size_t i_order = 1; i_order < BDFBaseType::mOrder + 1; ++i_order)
+                            un0[i_dim] -= BDFBaseType::mBDF[i_order] * it_node->FastGetSolutionStepValue(disp_components[i_dim], i_order);
+                        un0[i_dim] /= BDFBaseType::mBDF[i_dim];
+                        predicted[i_dim] = true;
+                    }
                 }
             }
-            if (disppos_z > -1 && !predicted_z) {
-                if (!it_node->GetDof(DISPLACEMENT_Z, disppos_z).IsFixed() && !predicted_z) {
-                    un0[2] = un1[2] + delta_time * dotun1[2] + 0.5 * std::pow(delta_time, 2) * dot2un1[2];
+            if (disppos > -1) {
```

Comment:
dispost cannot be -1 so this if is not needed

Reply:
Solved

Reply:
...it is still there. no need of this if.

Reply:
![imagen](https://user-images.githubusercontent.com/19991680/54746351-38363900-4bcc-11e9-9506-987698945088.png)

Outdated